rootston/desktop.c

```diff
@@ -200,6 +201,35 @@ struct roots_view *view_at(struct roots_desktop *desktop, double lx, double ly,
 	return NULL;
 }
 
+bool layer_surface_is_at(struct roots_desktop *desktop,
+		struct wlr_layer_surface *layer_surface, double lx, double ly,
+		double *sx, double *sy) {
+	struct wlr_box *output_box = wlr_output_layout_get_box(desktop->layout,
+		layer_surface->output);
+	double x, y;
+	wlr_layer_surface_get_position(layer_surface, &x, &y);
+	double layer_sx = lx - output_box->x - x;
+	double layer_sy = ly - output_box->y - y;
+
+	struct wlr_box box = {
+		.x = 0,
+		.y = 0,
+		.width = layer_surface->surface->current->buffer_width,
+		.height = layer_surface->surface->current->buffer_height,
```

Comment:
We make this mistake everywhere in the code and it's going to lead to a lot of bugs. But I think to actually get the width/height, you want to use `current->[width|height]` which takes into account the scale and transform of the buffer.